lib/puppet/type/cisco_portchannel_global.rb

```diff
@@ -0,0 +1,133 @@
+# Manages the Cisco PortChannel Global configuration resource.
+#
+# Dec 2015
+#
+# Copyright (c) 2013-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_portchannel_global) do
+  @doc = "
+    Manages the Cisco PortChannel Global configuration resource.
+    cisco_portchannel_global {'default':
+      ..attributes..
+    }
+    'default' is only acceptable name for this global config object.
+    Example:
+    cisco_portchannel_global { 'default':
+      asymmetric                     => 'true',
+      bundle_hash                    => 'ip',
+      bundle_select                  => 'dst',
+      concatenation                  => 'false',
+      hash_distribution              => 'adaptive',
+      hash_poly                      => 'CRC10a',
+      load_defer                     => '1000',
+      resilient                      => 'true',
+      rotate                         => '10',
+      symmetry                       => 'true',
+    }
+  "
+  ###################
+  # Resource Naming #
+  ###################
+
+  newparam(:name, namevar: :true) do
+    desc 'ID of the portchannel global config. Valid values are default.'
+
+    validate do |inst_name|
+      fail "only acceptable name is 'default'" if inst_name != 'default'
+    end
+  end # param id
+
+  ##############
+  # Attributes #
+  ##############
+
+  newproperty(:asymmetric) do
+    desc 'Asymmetric hash.'
+
+    newvalues(:true, :false, :default)
+  end # property asymmetric
+
+  newproperty(:bundle_hash) do
+    desc 'Bundle hash'
+
+    newvalues(:default, :ip, :'ip-l4port', :'ip-l4port-vlan',
```

Comment:
I changed my mind about adding platform support to the `desc` here.  Updating the README file should be sufficient.